### ISSUE TITLE
Discuss App Store rules in JavaScritCore article 

### ISSUE THREAD

**Issue by getaaron** [NSHipster/articles#252]
I suggest that the article should be expanded to address JavaScriptCore's relationship with App Store rules 2.6 and 2.7:

> 2.6 Apps that download code in any way or form will be rejected. 
> 2.7 Apps that install or launch other executable code will be rejected

These rules contradict this suggestion in the article:

> How can you use JavaScript in your apps? JavaScript snippets could be the basis for user-defined plugins that ship alongside yours.

I don't know whether this would be allowed.

Is there any history here? Have apps done this? What are the safety / security concerns? The article could be improved by discussing this contradiction.

**Comment by natecook1000** [comment]
I have a half-written email to you on this very topic! The suggestion in the article *is* a little blithe about the App Store policies, but the policy itself is more nuanced than that summary implies. From the iOS Developer Program License Agreement:

> **3.3.2** An Application may not download or install executable code. Interpreted code may only be used in an Application if all scripts, code and interpreters are packaged in the Application and not downloaded. The only exception to the foregoing is scripts and code downloaded and run by Apple's built-in WebKit framework, provided that such scripts and code do not change the primary purpose of the Application by providing features or functionality that are inconsistent with the intended and advertised purpose of the Application as submitted to the App Store.

Since JavaScriptCore is a part of WebKit, that would seem to exempt this use from the "no downloaded code" rule. That said, you're certainly correct that it's a gray, whim-of-the-reviewer area to be living in—I'll add a note to the article addressing it.

**Comment by croath** [comment]
I don't have any REAL evidences but just wanna say something I know.

According to what I've known that at the very first time Apple doesn't take an eye on these rules. During that period there were lots of apps(especially games) downloading executable Lua within apps. Then after that time Apple do obey these rules very strictly, these kinda apps could not be released anyway, it's about sometime maybe in 2011 or 2012. And finally Apple start to allow these operations for legal using of code.

What I wanna say is: You don't have a way to find if an app is downloading or running code that download from server. If you cannot determine it happened or not, then how can you forbid it?

Then, why did Apple make a rule? In my opinion, it's to forbid calling private methods in that way.

In other words, if you obey these rules, then there's no webview in your apps(b/s there will be javascript html css all around here!)
